src/main/java/org/elasticsearch/index/query/TermQueryBuilder.java

```diff
@@ -204,18 +240,18 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
             // move to the next token
             parser.nextToken();
         }
+    }
 
-        if (value == null) {
-            throw new QueryParsingException(parseContext.index(), "No value specified for term query");
-        }
-
+    public Query toQuery(QueryParseContext parseContext) {
+        Preconditions.checkNotNull(fieldName, "no fieldName specified for term query");
+        Preconditions.checkNotNull(value, "no value specified for term query");
         Query query = null;
         MapperService.SmartNameFieldMappers smartNameFieldMappers = parseContext.smartFieldMappers(fieldName);
```

Comment:
now I see why `QueryParseContext` here too. we should probably split the QueryParseContext in two data structures, one needed for parsing and one needed for toQuery (e.g. mapping etc.)